### DISCUSSION TITLE
Can't specify scripts path in ini file

### DISCUSSION THREAD

**Discussion by delboy1966** [ConSol-Monitoring/snclient#28] [General]
On NSClient I had some external scripts that lived in "c:\program files\NSClient++\scripts".
So on the snclient install I did the same thing, created a directory in c:\program files\snclient\scripts and dropped my external scripts in there.
But when running the check it can't find the scripts, my external command line is

**checktest = scripts\check_test.bat**

Which is exactly the same as I had it in the nsclient ini file, but when I run the check_nrpe command I get:

**UNKNOWN: proc: exec: "scripts\\check_test.bat": file does not exist**

So guessed I had to set the paths as

**; shared-path- Path to shared things.
shared-path = "c:\Program Files\snclient"

; scripts - Path to scripts.
scripts = ${shared-path}\scripts**

And amend my command line in the ini file to

**checktest = ${scripts}\check_test.bat**

But then get:

**UNKNOWN: proc: exec: "C:\\Program": file does not exist**

Have tried without quotes in the path with single quotes and with double quotes.

It works if i put my scripts in c:\scripts as there are no spaces in the path.
Am I not able to specify the path if it has spaces in?

**Comment by sni** [comment]
which version is that? Could you try the latest build from the actions?
https://github.com/ConSol-Monitoring/snclient/actions/workflows/cicd.yml

**Reply by delboy1966** [reply]
I downloaded the latest windows installer from here https://github.com/ConSol-Monitoring/snclient/releases/tag/v0.04

**Reply by sni** [reply]
right, but snclient is still under heavy development, so please try the latest build from the link above.

**Reply by delboy1966** [reply]
Ok, here is where I be honest, I'm not that familiar with GitHub and can't find an installer on that link.
I'm a little lost. Sorry.

**Reply by sni** [reply]
no worries, i just updated the docs here: https://github.com/ConSol-Monitoring/snclient/blob/main/docs/install/_index.md#development-snapshots

let me know if that works.

**Reply by delboy1966** [reply]
Cheers. I'll give it a go.

**Reply by delboy1966** [reply]
Sorry, been off ill last couple of days, so haven't had a chance to try it.
But I have done this morning, and still getting the same problem.
Must be something I'm doing wrong, with NSClient I didn't have to specify any paths, other in the check command and prepending the script name with "scripts/<script_name>" and it picked up the scripts from "C:\Program Files\NSClient++\scripts\".

How should the follow ini file lines look, so it just picks up the scripts from the scripts directory?

shared-path = 
scripts = 
checktest = check_test.bat

**Reply by delboy1966** [reply]
FYI
As a test, I set the command line as

**checktest = "C:\Program Files\snclient\scripts\check_test.bat"
checktest2 = 'C:\Program Files\snclient\scripts\check_test.bat'**

One with double quotes and one with single quotes. But still get this from running check_nrpe

**UNKNOWN: proc: exec: "C:\\Program": file does not exist**

Doesn't seem to like the spaces in the path, and doesn't work by enclosing it in quotes of any sort.
Its not a massive issue, as I can always change the path to the scripts to "c:\nagios_scripts", which seems to work.

**Reply by sni** [reply]
did it work like that before?

**Reply by delboy1966** [reply]
Yeah, when I first installed it and when I installed the latest version that you pointed me to.

**Reply by sni** [reply]
ok thanks, i need to investigate this.

**Reply by delboy1966** [reply]
Cheers, appreciate it.

**Reply by sni** [reply]
Did you by chance try if this is still an issue in the last release?
